[MSE] Run MediaSourcePrivateRemote and SourceBufferPrivateRemote in a dedicated work queue

jyavenard opened the pull request:
#### 90086fffd11ed100b343d615c03cf446a59734e0
<pre>
[MSE] Run MediaSourcePrivateRemote and SourceBufferPrivateRemote in a dedicated work queue
<a href="https://bugs.webkit.org/show_bug.cgi?id=265982">https://bugs.webkit.org/show_bug.cgi?id=265982</a>
<a href="https://rdar.apple.com/119296319">rdar://119296319</a>

Reviewed by Jer Noble.

Make the MediaSourcePrivateRemote and SourceBufferPrivateRemote use a WorkQueueMessageReceiver
with a dedicated WorkQueue.
The threading model in the GPU process remains unchanged as the current implementations
and relationship between SourceBufferPrivate and MediaPlayerPrivate rely
heavily on the objects being main-thread only (in particular the extensive use of
AVSampleBufferDisplayerLayer).
From an observer point of view in the content process, the various MSE objects operations
are now fully parallelised.

The threading flow as such is now:

Content Process:
MediaSource (main thread) -&gt; MediaSourcePrivateRemote (workqueue)
SourceBuffer (main thread) -&gt; SourceBufferPrivateRemote (workqueue)

GPU Process:
RemoteMediaSourceProxy (main thread) -&gt; MediaSourcePrivate (main thread)
RemoteSourceBufferProxy (main thread) -&gt; SourceBufferPrivate (main thread)

The MediaPlayerPrivate/MediaPlayerPrivateRemote runs on the main thread.

Some MediaSourcePrivate methods had to be made thread-safe to more easily synchronise member attributes.

Covered by existing tests, no change in observable behaviour.

* Source/WTF/wtf/Forward.h:
Add SerialFunctionDispatcher and RefCountedSerialFunctionDispatcher forward declaration.
* Source/WebCore/Modules/mediasource/ManagedMediaSource.cpp:
(WebCore::ManagedMediaSource::elementDetached):
(WebCore::ManagedMediaSource::setStreaming):
(WebCore::ManagedMediaSource::monitorSourceBuffers):
* Source/WebCore/Modules/mediasource/ManagedMediaSource.h:
* Source/WebCore/Modules/mediasource/MediaSource.cpp:
(WebCore::MediaSource::MediaSource):
(WebCore::MediaSource::waitForTarget):
(WebCore::MediaSource::completeSeek):
(WebCore::MediaSource::seekToTime):
(WebCore::MediaSource::seekable):
(WebCore::MediaSource::setLiveSeekableRange):
(WebCore::MediaSource::clearLiveSeekableRange):
(WebCore::MediaSource::hasBufferedTime):
(WebCore::MediaSource::hasFutureTime):
(WebCore::MediaSource::monitorSourceBuffers):
(WebCore::MediaSource::setDuration):
(WebCore::MediaSource::setDurationInternal):
(WebCore::MediaSource::setReadyState):
(WebCore::MediaSource::endOfStream):
(WebCore::MediaSource::streamEndedWithError):
(WebCore::MediaSource::addSourceBuffer):
(WebCore::MediaSource::detachFromElement):
(WebCore::MediaSource::onReadyStateChange):
(WebCore::MediaSource::activeRanges const):
(WebCore::MediaSource::regenerateActiveSourceBuffers):
(WebCore::MediaSource::notifyElementUpdateMediaState const):
(WebCore::MediaSource::sourceBufferBufferedChanged):
(WebCore::MediaSource::updateBufferedIfNeeded):
(WebCore::MediaSource::failedToCreateRenderer):
(WebCore::MediaSource::sourceBufferReceivedFirstInitializationSegmentChanged):
(WebCore::MediaSource::sourceBufferActiveTrackFlagChanged):
(WebCore::MediaSource::setMediaPlayerReadyState):
(WebCore::MediaSource::memoryPressure):
(WebCore::MediaSource::ensureWeakOnDispatcher const):
* Source/WebCore/Modules/mediasource/MediaSource.h:
(WebCore::MediaSource::elementDetached):
(WebCore::MediaSource::promisedWeakOnDispatcher const):
* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::m_logIdentifier):
(WebCore::SourceBuffer::~SourceBuffer):
(WebCore::SourceBuffer::rangeRemoval):
(WebCore::SourceBuffer::removedFromMediaSource):
(WebCore::SourceBuffer::seekToTime):
(WebCore::SourceBuffer::appendBufferInternal):
(WebCore::SourceBuffer::sourceBufferPrivateDidReceiveRenderingError):
(WebCore::SourceBuffer::maximumBufferSize const):
(WebCore::SourceBuffer::setActive):
(WebCore::SourceBuffer::sourceBufferPrivateDidReceiveInitializationSegment):
(WebCore::SourceBuffer::sourceBufferPrivateDurationChanged):
(WebCore::SourceBuffer::sourceBufferPrivateHighestPresentationTimestampChanged):
(WebCore::SourceBuffer::sourceBufferPrivateDidDropSample):
(WebCore::SourceBuffer::sourceBufferPrivateBufferedChanged):
(WebCore::SourceBuffer::ensureWeakOnDispatcher const):
(WebCore::SourceBuffer::promisedWeakOnDispatcher const):
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/platform/graphics/MediaSourcePrivate.cpp:
(WebCore::MediaSourcePrivate::MediaSourcePrivate):
(WebCore::MediaSourcePrivate::duration const):
(WebCore::MediaSourcePrivate::removeSourceBuffer):
(WebCore::MediaSourcePrivate::sourceBufferPrivateDidChangeActiveState):
(WebCore::MediaSourcePrivate::hasAudio const):
(WebCore::MediaSourcePrivate::hasVideo const):
(WebCore::MediaSourcePrivate::durationChanged):
(WebCore::MediaSourcePrivate::bufferedChanged):
(WebCore::MediaSourcePrivate::buffered const):
(WebCore::MediaSourcePrivate::setCDMSession):
(WebCore::MediaSourcePrivate::ensureOnDispatcher const):
* Source/WebCore/platform/graphics/MediaSourcePrivate.h:
* Source/WebCore/platform/graphics/MediaSourcePrivateClient.h:
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::SourceBufferPrivate::SourceBufferPrivate):
(WebCore::SourceBufferPrivate::removedFromMediaSource):
(WebCore::SourceBufferPrivate::setClient):
(WebCore::SourceBufferPrivate::updateBuffered):
(WebCore::SourceBufferPrivate::seekToTime):
(WebCore::SourceBufferPrivate::removeCodedFrames):
(WebCore::SourceBufferPrivate::updateTrackIds):
(WebCore::SourceBufferPrivate::setAllTrackBuffersNeedRandomAccess):
(WebCore::SourceBufferPrivate::didReceiveInitializationSegment):
(WebCore::SourceBufferPrivate::didUpdateFormatDescriptionForTrackId):
(WebCore::SourceBufferPrivate::append):
(WebCore::SourceBufferPrivate::resetParserState):
(WebCore::SourceBufferPrivate::memoryPressure):
(WebCore::SourceBufferPrivate::ensureOnDispatcher const):
(WebCore::SourceBufferPrivate::detach): Deleted.
* Source/WebCore/platform/graphics/SourceBufferPrivate.h:
* Source/WebCore/platform/graphics/SourceBufferPrivateClient.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.cpp:
(WebKit::RemoteMediaPlayerProxy::commitAllTransactions):
(WebKit::RemoteMediaPlayerProxy::mediaPlayerNetworkStateChanged):
(WebKit::RemoteMediaPlayerProxy::mediaPlayerReadyStateChanged):
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.h:
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.cpp:
(WebKit::RemoteMediaSourceProxy::waitForTarget):
(WebKit::RemoteMediaSourceProxy::seekToTime):
(WebKit::RemoteMediaSourceProxy::shutdown):
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.cpp:
(WebKit::RemoteSourceBufferProxy::~RemoteSourceBufferProxy):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateDidReceiveInitializationSegment):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateHighestPresentationTimestampChanged):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateDurationChanged):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateBufferedChanged):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateDidDropSample):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateDidReceiveRenderingError):
(WebKit::RemoteSourceBufferProxy::append):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/GPUProcessConnection.cpp:
(WebKit::GPUProcessConnection::dispatchMessage):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::setReadyState):
(WebKit::MediaPlayerPrivateRemote::readyStateChanged):
(WebKit::MediaPlayerPrivateRemote::updateCachedState):
(WebKit::MediaPlayerPrivateRemote::addRemoteAudioTrack):
(WebKit::MediaPlayerPrivateRemote::removeRemoteAudioTrack):
(WebKit::MediaPlayerPrivateRemote::remoteAudioTrackConfigurationChanged):
(WebKit::MediaPlayerPrivateRemote::addRemoteTextTrack):
(WebKit::MediaPlayerPrivateRemote::removeRemoteTextTrack):
(WebKit::MediaPlayerPrivateRemote::remoteTextTrackConfigurationChanged):
(WebKit::MediaPlayerPrivateRemote::parseWebVTTFileHeader):
(WebKit::MediaPlayerPrivateRemote::parseWebVTTCueData):
(WebKit::MediaPlayerPrivateRemote::parseWebVTTCueDataStruct):
(WebKit::MediaPlayerPrivateRemote::addDataCue):
(WebKit::MediaPlayerPrivateRemote::addDataCueWithType):
(WebKit::MediaPlayerPrivateRemote::updateDataCue):
(WebKit::MediaPlayerPrivateRemote::removeDataCue):
(WebKit::MediaPlayerPrivateRemote::addGenericCue):
(WebKit::MediaPlayerPrivateRemote::updateGenericCue):
(WebKit::MediaPlayerPrivateRemote::removeGenericCue):
(WebKit::MediaPlayerPrivateRemote::addRemoteVideoTrack):
(WebKit::MediaPlayerPrivateRemote::removeRemoteVideoTrack):
(WebKit::MediaPlayerPrivateRemote::remoteVideoTrackConfigurationChanged):
(WebKit::MediaPlayerPrivateRemote::requestResource):
(WebKit::MediaPlayerPrivateRemote::removeResource):
(WebKit::MediaPlayerPrivateRemote::audioTrackPrivateRemote const):
(WebKit::MediaPlayerPrivateRemote::videoTrackPrivateRemote const):
(WebKit::MediaPlayerPrivateRemote::textTrackPrivateRemote const):
(WebKit::MediaPlayerPrivateRemote::commitAllTransactions):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.messages.in:
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
(WebKit::MediaSourcePrivateRemote::queue):
(WebKit::MediaSourcePrivateRemote::ensureOnDispatcherSync const):
(WebKit::MediaSourcePrivateRemote::MediaSourcePrivateRemote):
(WebKit::MediaSourcePrivateRemote::~MediaSourcePrivateRemote):
(WebKit::MediaSourcePrivateRemote::addSourceBuffer):
(WebKit::MediaSourcePrivateRemote::durationChanged):
Directly notify the MediaPlayerPrivateRemote that the duration as changed
rather than go through two round-trips to the GPU process.
(WebKit::MediaSourcePrivateRemote::bufferedChanged):
(WebKit::MediaSourcePrivateRemote::markEndOfStream):
(WebKit::MediaSourcePrivateRemote::unmarkEndOfStream):
(WebKit::MediaSourcePrivateRemote::setMediaPlayerReadyState):
Directly notify the MediaPlayerPrivateRemote that the readyState as changed
rather than go through two round-trips to the GPU process.
(WebKit::MediaSourcePrivateRemote::setTimeFudgeFactor):
(WebKit::MediaSourcePrivateRemote::MessageReceiver::client const):
(WebKit::MediaSourcePrivateRemote::MessageReceiver::proxyWaitForTarget):
(WebKit::MediaSourcePrivateRemote::MessageReceiver::proxySeekToTime):
(WebKit::MediaSourcePrivateRemote::MessageReceiver::mediaSourcePrivateShuttingDown):
(WebKit::MediaSourcePrivateRemote::MessageReceiver::MessageReceiver):
(WebKit::MediaSourcePrivateRemote::proxyWaitForTarget): Deleted.
(WebKit::MediaSourcePrivateRemote::proxySeekToTime): Deleted.
(WebKit::MediaSourcePrivateRemote::mediaSourcePrivateShuttingDown): Deleted.
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemoteMessageReceiver.h: Copied from Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.messages.in.
The current infrastructure doesn&apos;t allow to define the IPC protocol in a private sub-class, so add an alias.
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemoteMessageReceiver.messages.in: Copied from Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.messages.in.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerState.h:
Remove readyState from being state caching structure. As the readyState was updated alongside other attributes, it was possible
for the readyState to be upgraded with a GPU&apos;s stale value before updated to the new value from the content process. Leading to intermittent
failures, that those changes made more likely.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerState.serialization.in:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
(WebKit::SourceBufferPrivateRemote::append):
(WebKit::SourceBufferPrivateRemote::isActive const):
(WebKit::SourceBufferPrivateRemote::setActive):
(WebKit::SourceBufferPrivateRemote::setMediaSourceEnded):
(WebKit::SourceBufferPrivateRemote::queue):
(WebKit::SourceBufferPrivateRemote::ensureOnDispatcherSync):
(WebKit::SourceBufferPrivateRemote::SourceBufferPrivateRemote):
(WebKit::SourceBufferPrivateRemote::~SourceBufferPrivateRemote):
(WebKit::SourceBufferPrivateRemote::append):
(WebKit::SourceBufferPrivateRemote::abort):
(WebKit::SourceBufferPrivateRemote::resetParserState):
(WebKit::SourceBufferPrivateRemote::removedFromMediaSource):
(WebKit::SourceBufferPrivateRemote::canSwitchToType):
(WebKit::SourceBufferPrivateRemote::setMediaSourceEnded):
(WebKit::SourceBufferPrivateRemote::setMode):
(WebKit::SourceBufferPrivateRemote::removeCodedFrames):
(WebKit::SourceBufferPrivateRemote::evictCodedFrames):
(WebKit::SourceBufferPrivateRemote::addTrackBuffer):
(WebKit::SourceBufferPrivateRemote::resetTrackBuffers):
(WebKit::SourceBufferPrivateRemote::clearTrackBuffers):
(WebKit::SourceBufferPrivateRemote::setAllTrackBuffersNeedRandomAccess):
(WebKit::SourceBufferPrivateRemote::setGroupStartTimestamp):
(WebKit::SourceBufferPrivateRemote::setGroupStartTimestampToEndTimestamp):
(WebKit::SourceBufferPrivateRemote::setShouldGenerateTimestamps):
(WebKit::SourceBufferPrivateRemote::reenqueueMediaIfNeeded):
(WebKit::SourceBufferPrivateRemote::resetTimestampOffsetInTrackBuffers):
(WebKit::SourceBufferPrivateRemote::startChangingType):
(WebKit::SourceBufferPrivateRemote::timestampOffset const):
(WebKit::SourceBufferPrivateRemote::setTimestampOffset):
(WebKit::SourceBufferPrivateRemote::setAppendWindowStart):
(WebKit::SourceBufferPrivateRemote::setAppendWindowEnd):
(WebKit::SourceBufferPrivateRemote::computeSeekTime):
(WebKit::SourceBufferPrivateRemote::seekToTime):
(WebKit::SourceBufferPrivateRemote::updateTrackIds):
(WebKit::SourceBufferPrivateRemote::bufferedSamplesForTrackId):
(WebKit::SourceBufferPrivateRemote::enqueuedSamplesForTrackID):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::client const):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::takeOwnershipOfMemory):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateDidReceiveInitializationSegment):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateHighestPresentationTimestampChanged):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateDurationChanged):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateBufferedChanged):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateDidDropSample):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateDidReceiveRenderingError):
(WebKit::SourceBufferPrivateRemote::memoryPressure):
(WebKit::SourceBufferPrivateRemote::minimumUpcomingPresentationTimeForTrackID):
(WebKit::SourceBufferPrivateRemote::setMaximumQueueDepthForTrackID):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::MessageReceiver):
(WebKit::SourceBufferPrivateRemote::takeOwnershipOfMemory): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateDidReceiveInitializationSegment): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateHighestPresentationTimestampChanged): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateDurationChanged): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateBufferedChanged): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateDidDropSample): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateDidReceiveRenderingError): Deleted.
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemoteMessageReceiver.h: Renamed from Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.messages.in.
The current infrastructure doesn&apos;t allow to define the IPC protocol in a private sub-class, so add an alias.
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemoteMessageReceiver.messages.in: Renamed from Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.messages.in.

Canonical link: <a href="https://commits.webkit.org/274356@main">https://commits.webkit.org/274356@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8303878cb9531227efa24622505e493997fab09c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38818 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17749 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41163 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41349 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/34471 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/41125 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/20649 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15098 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32543 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39391 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14935 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33676 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12979 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12937 "8 flakes 5 failures") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/34580 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42625 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/32264 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35232 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34939 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38765 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/38441 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13628 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11241 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36981 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15240 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/45450 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8696 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/14903 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/9267 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/14721 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jernoble commented on Source/WebCore/Modules/mediasource/ManagedMediaSource.cpp:
```diff
@@ -132,35 +134,39 @@ void ManagedMediaSource::ensurePrefsRead()
-    }
-    auto currentTime = this->currentTime();
+        if (!protectedThis || isClosed()) {
+            setStreaming(false);
```
This will cause a crash if `weakThis` (and therefore `protectedThis`) is null.

jyavenard replied:
doh

jernoble commented on Source/WebCore/Modules/mediasource/MediaSource.cpp:
```diff
@@ -1300,19 +1354,32 @@ void MediaSource::sourceBufferActiveTrackFlagChanged(bool activeTrackFlag)
+    }
+    m_dispatcher->dispatch(WTFMove(function));
+}
+
```
I wonder if it would be useful for `ensureOnDispatcher()` to do the `weakThis`->`protectedThis` itself, and do the null check before calling the function. If you think this is valuable, you may need to rename it, something like `ensureProtectedOnDispatcher()` or `ensureWeakOnDispatcher()`.

jyavenard replied:
done

jernoble commented on Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
```diff
@@ -664,257 +680,260 @@ void SourceBuffer::setActive(bool active)
-    }
+    return invokeAsync(m_dispatcher, [weakThis = ThreadSafeWeakPtr { *this }, this, segment = WTFMove(segment)]() mutable {
+        RefPtr protectedThis = weakThis.get();
+        if (!protectedThis || isRemoved())
```
Seems like this is `ensureOnDispatcher()` but returns a Promise. Should there be a `ensureOnDispatcher()` variant that similarly returns a promise?.

jyavenard replied:
invokeAsync is a templated utility that returns the same type as what the lambda returns. If we only ever returned the same promise type that would do, but we aren't.

This code will have to be changed in a later PR anyway as invokeAsync requires a guarantee that the invoked lambda gets called, which once we run on a worker will not happen.

jernoble commented on Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
```diff
@@ -151,50 +189,72 @@ MediaPlayer::ReadyState MediaSourcePrivateRemote::mediaPlayerReadyState() const
+            return;
+        }
+    }
+    completionHandler(makeUnexpected(PlatformMediaError::ClientDisconnected));
```
Style nit: prefer early returns to nested ifs.

jernoble commented on Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
```diff
@@ -151,50 +189,72 @@ MediaPlayer::ReadyState MediaSourcePrivateRemote::mediaPlayerReadyState() const
+            return;
+        }
+    }
+    completionHandler(makeUnexpected(PlatformMediaError::SourceRemoved));
```
Ditto.

jernoble commented on Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
```diff
@@ -151,50 +189,72 @@ MediaPlayer::ReadyState MediaSourcePrivateRemote::mediaPlayerReadyState() const
-        downcast<SourceBufferPrivateRemote>(sourceBuffer)->disconnect();
+    assertIsCurrent(MediaSourcePrivateRemote::queue());
+
+    if (RefPtr parent = m_parent.get()) {
+        parent->m_shutdown = true;
+        parent->m_readyState = MediaPlayer::ReadyState::HaveNothing;
+        for (auto& sourceBuffer : parent->m_sourceBuffers)
+            downcast<SourceBufferPrivateRemote>(sourceBuffer)->disconnect();
+    };
     completionHandler();
```
Ditto.

jernoble commented on Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
```diff
@@ -135,332 +145,415 @@ void SourceBufferPrivateRemote::resetParserStateInternal()
+            client->sourceBufferPrivateHighestPresentationTimestampChanged(timestamp);
+    }
 }
 
```
Ditto.

jernoble commented on Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
```diff
@@ -135,332 +145,415 @@ void SourceBufferPrivateRemote::resetParserStateInternal()
+            client->sourceBufferPrivateDurationChanged(duration)->whenSettled(SourceBufferPrivateRemote::queue(), WTFMove(completionHandler));
+            return;
+        }
+    }
```
Ditto. This pattern shows up enough, I wonder if it needs a helper `client()` method.

jernoble commented on Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
```diff
@@ -135,332 +145,415 @@ void SourceBufferPrivateRemote::resetParserStateInternal()
+            client->sourceBufferPrivateBufferedChanged(WTFMove(trackBuffersRanges), extraMemory)->whenSettled(SourceBufferPrivateRemote::queue(), WTFMove(completionHandler));
+            return;
+        }
+    }
```
Ditto.

jernoble commented on Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
```diff
@@ -135,332 +145,415 @@ void SourceBufferPrivateRemote::resetParserStateInternal()
+        if (auto client = parent->client())
+            client->sourceBufferPrivateDidDropSample();
+    }
 }
```
Ditto.

jyavenard committed:
[MSE] Run MediaSourcePrivateRemote and SourceBufferPrivateRemote in a dedicated work queue
https://bugs.webkit.org/show_bug.cgi?id=265982
rdar://119296319

Reviewed by Jer Noble.

Make the MediaSourcePrivateRemote and SourceBufferPrivateRemote use a WorkQueueMessageReceiver
with a dedicated WorkQueue.
The threading model in the GPU process remains unchanged as the current implementations
and relationship between SourceBufferPrivate and MediaPlayerPrivate rely
heavily on the objects being main-thread only (in particular the extensive use of
AVSampleBufferDisplayerLayer).
From an observer point of view in the content process, the various MSE objects operations
are now fully parallelised.

The threading flow as such is now:

Content Process:
MediaSource (main thread) -> MediaSourcePrivateRemote (workqueue)
SourceBuffer (main thread) -> SourceBufferPrivateRemote (workqueue)

GPU Process:
RemoteMediaSourceProxy (main thread) -> MediaSourcePrivate (main thread)
RemoteSourceBufferProxy (main thread) -> SourceBufferPrivate (main thread)

The MediaPlayerPrivate/MediaPlayerPrivateRemote runs on the main thread.

Some MediaSourcePrivate methods had to be made thread-safe to more easily synchronise member attributes.

Covered by existing tests, no change in observable behaviour.

* Source/WTF/wtf/Forward.h:
Add SerialFunctionDispatcher and RefCountedSerialFunctionDispatcher forward declaration.
* Source/WebCore/Modules/mediasource/ManagedMediaSource.cpp:
(WebCore::ManagedMediaSource::elementDetached):
(WebCore::ManagedMediaSource::setStreaming):
(WebCore::ManagedMediaSource::monitorSourceBuffers):
* Source/WebCore/Modules/mediasource/ManagedMediaSource.h:
* Source/WebCore/Modules/mediasource/MediaSource.cpp:
(WebCore::MediaSource::MediaSource):
(WebCore::MediaSource::waitForTarget):
(WebCore::MediaSource::completeSeek):
(WebCore::MediaSource::seekToTime):
(WebCore::MediaSource::seekable):
(WebCore::MediaSource::setLiveSeekableRange):
(WebCore::MediaSource::clearLiveSeekableRange):
(WebCore::MediaSource::hasBufferedTime):
(WebCore::MediaSource::hasFutureTime):
(WebCore::MediaSource::monitorSourceBuffers):
(WebCore::MediaSource::setDuration):
(WebCore::MediaSource::setDurationInternal):
(WebCore::MediaSource::setReadyState):
(WebCore::MediaSource::endOfStream):
(WebCore::MediaSource::streamEndedWithError):
(WebCore::MediaSource::addSourceBuffer):
(WebCore::MediaSource::detachFromElement):
(WebCore::MediaSource::onReadyStateChange):
(WebCore::MediaSource::activeRanges const):
(WebCore::MediaSource::regenerateActiveSourceBuffers):
(WebCore::MediaSource::notifyElementUpdateMediaState const):
(WebCore::MediaSource::sourceBufferBufferedChanged):
(WebCore::MediaSource::updateBufferedIfNeeded):
(WebCore::MediaSource::failedToCreateRenderer):
(WebCore::MediaSource::sourceBufferReceivedFirstInitializationSegmentChanged):
(WebCore::MediaSource::sourceBufferActiveTrackFlagChanged):
(WebCore::MediaSource::setMediaPlayerReadyState):
(WebCore::MediaSource::memoryPressure):
(WebCore::MediaSource::ensureWeakOnDispatcher const):
* Source/WebCore/Modules/mediasource/MediaSource.h:
(WebCore::MediaSource::elementDetached):
(WebCore::MediaSource::promisedWeakOnDispatcher const):
* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::m_logIdentifier):
(WebCore::SourceBuffer::~SourceBuffer):
(WebCore::SourceBuffer::rangeRemoval):
(WebCore::SourceBuffer::removedFromMediaSource):
(WebCore::SourceBuffer::seekToTime):
(WebCore::SourceBuffer::appendBufferInternal):
(WebCore::SourceBuffer::sourceBufferPrivateDidReceiveRenderingError):
(WebCore::SourceBuffer::maximumBufferSize const):
(WebCore::SourceBuffer::setActive):
(WebCore::SourceBuffer::sourceBufferPrivateDidReceiveInitializationSegment):
(WebCore::SourceBuffer::sourceBufferPrivateDurationChanged):
(WebCore::SourceBuffer::sourceBufferPrivateHighestPresentationTimestampChanged):
(WebCore::SourceBuffer::sourceBufferPrivateDidDropSample):
(WebCore::SourceBuffer::sourceBufferPrivateBufferedChanged):
(WebCore::SourceBuffer::ensureWeakOnDispatcher const):
(WebCore::SourceBuffer::promisedWeakOnDispatcher const):
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/platform/graphics/MediaSourcePrivate.cpp:
(WebCore::MediaSourcePrivate::MediaSourcePrivate):
(WebCore::MediaSourcePrivate::duration const):
(WebCore::MediaSourcePrivate::removeSourceBuffer):
(WebCore::MediaSourcePrivate::sourceBufferPrivateDidChangeActiveState):
(WebCore::MediaSourcePrivate::hasAudio const):
(WebCore::MediaSourcePrivate::hasVideo const):
(WebCore::MediaSourcePrivate::durationChanged):
(WebCore::MediaSourcePrivate::bufferedChanged):
(WebCore::MediaSourcePrivate::buffered const):
(WebCore::MediaSourcePrivate::setCDMSession):
(WebCore::MediaSourcePrivate::ensureOnDispatcher const):
* Source/WebCore/platform/graphics/MediaSourcePrivate.h:
* Source/WebCore/platform/graphics/MediaSourcePrivateClient.h:
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::SourceBufferPrivate::SourceBufferPrivate):
(WebCore::SourceBufferPrivate::removedFromMediaSource):
(WebCore::SourceBufferPrivate::setClient):
(WebCore::SourceBufferPrivate::updateBuffered):
(WebCore::SourceBufferPrivate::seekToTime):
(WebCore::SourceBufferPrivate::removeCodedFrames):
(WebCore::SourceBufferPrivate::updateTrackIds):
(WebCore::SourceBufferPrivate::setAllTrackBuffersNeedRandomAccess):
(WebCore::SourceBufferPrivate::didReceiveInitializationSegment):
(WebCore::SourceBufferPrivate::didUpdateFormatDescriptionForTrackId):
(WebCore::SourceBufferPrivate::append):
(WebCore::SourceBufferPrivate::resetParserState):
(WebCore::SourceBufferPrivate::memoryPressure):
(WebCore::SourceBufferPrivate::ensureOnDispatcher const):
(WebCore::SourceBufferPrivate::detach): Deleted.
* Source/WebCore/platform/graphics/SourceBufferPrivate.h:
* Source/WebCore/platform/graphics/SourceBufferPrivateClient.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.cpp:
(WebKit::RemoteMediaPlayerProxy::commitAllTransactions):
(WebKit::RemoteMediaPlayerProxy::mediaPlayerNetworkStateChanged):
(WebKit::RemoteMediaPlayerProxy::mediaPlayerReadyStateChanged):
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.h:
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.cpp:
(WebKit::RemoteMediaSourceProxy::waitForTarget):
(WebKit::RemoteMediaSourceProxy::seekToTime):
(WebKit::RemoteMediaSourceProxy::shutdown):
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.cpp:
(WebKit::RemoteSourceBufferProxy::~RemoteSourceBufferProxy):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateDidReceiveInitializationSegment):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateHighestPresentationTimestampChanged):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateDurationChanged):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateBufferedChanged):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateDidDropSample):
(WebKit::RemoteSourceBufferProxy::sourceBufferPrivateDidReceiveRenderingError):
(WebKit::RemoteSourceBufferProxy::append):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/GPUProcessConnection.cpp:
(WebKit::GPUProcessConnection::dispatchMessage):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::setReadyState):
(WebKit::MediaPlayerPrivateRemote::readyStateChanged):
(WebKit::MediaPlayerPrivateRemote::updateCachedState):
(WebKit::MediaPlayerPrivateRemote::addRemoteAudioTrack):
(WebKit::MediaPlayerPrivateRemote::removeRemoteAudioTrack):
(WebKit::MediaPlayerPrivateRemote::remoteAudioTrackConfigurationChanged):
(WebKit::MediaPlayerPrivateRemote::addRemoteTextTrack):
(WebKit::MediaPlayerPrivateRemote::removeRemoteTextTrack):
(WebKit::MediaPlayerPrivateRemote::remoteTextTrackConfigurationChanged):
(WebKit::MediaPlayerPrivateRemote::parseWebVTTFileHeader):
(WebKit::MediaPlayerPrivateRemote::parseWebVTTCueData):
(WebKit::MediaPlayerPrivateRemote::parseWebVTTCueDataStruct):
(WebKit::MediaPlayerPrivateRemote::addDataCue):
(WebKit::MediaPlayerPrivateRemote::addDataCueWithType):
(WebKit::MediaPlayerPrivateRemote::updateDataCue):
(WebKit::MediaPlayerPrivateRemote::removeDataCue):
(WebKit::MediaPlayerPrivateRemote::addGenericCue):
(WebKit::MediaPlayerPrivateRemote::updateGenericCue):
(WebKit::MediaPlayerPrivateRemote::removeGenericCue):
(WebKit::MediaPlayerPrivateRemote::addRemoteVideoTrack):
(WebKit::MediaPlayerPrivateRemote::removeRemoteVideoTrack):
(WebKit::MediaPlayerPrivateRemote::remoteVideoTrackConfigurationChanged):
(WebKit::MediaPlayerPrivateRemote::requestResource):
(WebKit::MediaPlayerPrivateRemote::removeResource):
(WebKit::MediaPlayerPrivateRemote::audioTrackPrivateRemote const):
(WebKit::MediaPlayerPrivateRemote::videoTrackPrivateRemote const):
(WebKit::MediaPlayerPrivateRemote::textTrackPrivateRemote const):
(WebKit::MediaPlayerPrivateRemote::commitAllTransactions):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.messages.in:
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
(WebKit::MediaSourcePrivateRemote::queue):
(WebKit::MediaSourcePrivateRemote::ensureOnDispatcherSync const):
(WebKit::MediaSourcePrivateRemote::MediaSourcePrivateRemote):
(WebKit::MediaSourcePrivateRemote::~MediaSourcePrivateRemote):
(WebKit::MediaSourcePrivateRemote::addSourceBuffer):
(WebKit::MediaSourcePrivateRemote::durationChanged):
Directly notify the MediaPlayerPrivateRemote that the duration as changed
rather than go through two round-trips to the GPU process.
(WebKit::MediaSourcePrivateRemote::bufferedChanged):
(WebKit::MediaSourcePrivateRemote::markEndOfStream):
(WebKit::MediaSourcePrivateRemote::unmarkEndOfStream):
(WebKit::MediaSourcePrivateRemote::setMediaPlayerReadyState):
Directly notify the MediaPlayerPrivateRemote that the readyState as changed
rather than go through two round-trips to the GPU process.
(WebKit::MediaSourcePrivateRemote::setTimeFudgeFactor):
(WebKit::MediaSourcePrivateRemote::MessageReceiver::client const):
(WebKit::MediaSourcePrivateRemote::MessageReceiver::proxyWaitForTarget):
(WebKit::MediaSourcePrivateRemote::MessageReceiver::proxySeekToTime):
(WebKit::MediaSourcePrivateRemote::MessageReceiver::mediaSourcePrivateShuttingDown):
(WebKit::MediaSourcePrivateRemote::MessageReceiver::MessageReceiver):
(WebKit::MediaSourcePrivateRemote::proxyWaitForTarget): Deleted.
(WebKit::MediaSourcePrivateRemote::proxySeekToTime): Deleted.
(WebKit::MediaSourcePrivateRemote::mediaSourcePrivateShuttingDown): Deleted.
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemoteMessageReceiver.h: Copied from Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.messages.in.
The current infrastructure doesn't allow to define the IPC protocol in a private sub-class, so add an alias.
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemoteMessageReceiver.messages.in: Copied from Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.messages.in.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerState.h:
Remove readyState from being state caching structure. As the readyState was updated alongside other attributes, it was possible
for the readyState to be upgraded with a GPU's stale value before updated to the new value from the content process. Leading to intermittent
failures, that those changes made more likely.
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerState.serialization.in:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
(WebKit::SourceBufferPrivateRemote::append):
(WebKit::SourceBufferPrivateRemote::isActive const):
(WebKit::SourceBufferPrivateRemote::setActive):
(WebKit::SourceBufferPrivateRemote::setMediaSourceEnded):
(WebKit::SourceBufferPrivateRemote::queue):
(WebKit::SourceBufferPrivateRemote::ensureOnDispatcherSync):
(WebKit::SourceBufferPrivateRemote::SourceBufferPrivateRemote):
(WebKit::SourceBufferPrivateRemote::~SourceBufferPrivateRemote):
(WebKit::SourceBufferPrivateRemote::append):
(WebKit::SourceBufferPrivateRemote::abort):
(WebKit::SourceBufferPrivateRemote::resetParserState):
(WebKit::SourceBufferPrivateRemote::removedFromMediaSource):
(WebKit::SourceBufferPrivateRemote::canSwitchToType):
(WebKit::SourceBufferPrivateRemote::setMediaSourceEnded):
(WebKit::SourceBufferPrivateRemote::setMode):
(WebKit::SourceBufferPrivateRemote::removeCodedFrames):
(WebKit::SourceBufferPrivateRemote::evictCodedFrames):
(WebKit::SourceBufferPrivateRemote::addTrackBuffer):
(WebKit::SourceBufferPrivateRemote::resetTrackBuffers):
(WebKit::SourceBufferPrivateRemote::clearTrackBuffers):
(WebKit::SourceBufferPrivateRemote::setAllTrackBuffersNeedRandomAccess):
(WebKit::SourceBufferPrivateRemote::setGroupStartTimestamp):
(WebKit::SourceBufferPrivateRemote::setGroupStartTimestampToEndTimestamp):
(WebKit::SourceBufferPrivateRemote::setShouldGenerateTimestamps):
(WebKit::SourceBufferPrivateRemote::reenqueueMediaIfNeeded):
(WebKit::SourceBufferPrivateRemote::resetTimestampOffsetInTrackBuffers):
(WebKit::SourceBufferPrivateRemote::startChangingType):
(WebKit::SourceBufferPrivateRemote::timestampOffset const):
(WebKit::SourceBufferPrivateRemote::setTimestampOffset):
(WebKit::SourceBufferPrivateRemote::setAppendWindowStart):
(WebKit::SourceBufferPrivateRemote::setAppendWindowEnd):
(WebKit::SourceBufferPrivateRemote::computeSeekTime):
(WebKit::SourceBufferPrivateRemote::seekToTime):
(WebKit::SourceBufferPrivateRemote::updateTrackIds):
(WebKit::SourceBufferPrivateRemote::bufferedSamplesForTrackId):
(WebKit::SourceBufferPrivateRemote::enqueuedSamplesForTrackID):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::client const):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::takeOwnershipOfMemory):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateDidReceiveInitializationSegment):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateHighestPresentationTimestampChanged):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateDurationChanged):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateBufferedChanged):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateDidDropSample):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::sourceBufferPrivateDidReceiveRenderingError):
(WebKit::SourceBufferPrivateRemote::memoryPressure):
(WebKit::SourceBufferPrivateRemote::minimumUpcomingPresentationTimeForTrackID):
(WebKit::SourceBufferPrivateRemote::setMaximumQueueDepthForTrackID):
(WebKit::SourceBufferPrivateRemote::MessageReceiver::MessageReceiver):
(WebKit::SourceBufferPrivateRemote::takeOwnershipOfMemory): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateDidReceiveInitializationSegment): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateHighestPresentationTimestampChanged): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateDurationChanged): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateBufferedChanged): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateDidDropSample): Deleted.
(WebKit::SourceBufferPrivateRemote::sourceBufferPrivateDidReceiveRenderingError): Deleted.
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemoteMessageReceiver.h: Renamed from Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.messages.in.
The current infrastructure doesn't allow to define the IPC protocol in a private sub-class, so add an alias.
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemoteMessageReceiver.messages.in: Renamed from Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.messages.in.

Canonical link: https://commits.webkit.org/274356@main
